modules/library/main/src/main/java/org/geotools/data/util/NumericConverterFactory.java

```diff
@@ -325,6 +327,34 @@ static String toIntegral(String s) {
         }
     }
 
+    /**
+     * Clean the source handling possible edge cases.
+     *
+     * @return the cleaned source
+     */
+    static Object cleanSource(Object source) {
+        BigDecimal bigDecimal = getBigDecimalFromScientificNotation(source);
+        return bigDecimal != null ? bigDecimal : source;
+    }
+
+    /**
+     * @return a BigDecimal in case the source is a String in scientific notation, null
+     *     otherwise
+     */
+    static BigDecimal getBigDecimalFromScientificNotation(Object source) {
+        if (source instanceof String) {
+            try {
+                return ((String) source).toUpperCase().contains("E")
+                        ? new BigDecimal((String) source)
+                        : null;
+            } catch (NumberFormatException ex) {
```

Comment:
Can we at least log to finest that this failed - otherwise you get no clue as to what went wrong with out opening the debugger up.